src/App/Application.cpp

```diff
@@ -2281,27 +2281,64 @@ std::string Application::FindHomePath(const char* sCall)
 
 std::string Application::FindHomePath(const char* call)
 {
-    uint32_t sz = 0;
-    char *buf;
+    if (Py_IsInitialized()) {
+        // If we're initialised from an external Python, then there should be
+        // a path in the interpreter's sys.path that gets to FreeCAD.so .
+
+        auto pySysPath(PySys_GetObject("path"));
+        if ( PyList_Check(pySysPath) ) { 
+            // pySysPath should be a *PyList of strings - iterate through it
+            // backwards since the FreeCAD path was likely appended just before
+            // we were imported.
+            for (auto i( PyList_Size(pySysPath)-1 ); i >= 0 ; --i) {
+                auto pyPath( PyList_GetItem(pySysPath, i) );
+
+#if PY_MAJOR_VERSION >= 3
+                if ( !PyUnicode_Check(pyPath) )
+                    continue;
+                std::string path( PyUnicode_AsUTF8(pyPath) );
+#else 
+                if ( !PyString_Check(pyPath) )
+                    continue;
+                std::string path( PyString_AsString(pyPath) );
+#endif // #if/else PY_MAJOR_VERSION >= 3

```

Comment:
maybe checking for PyUnicode (python2 and 3)? and elif checking for PyString for python2?
````
    if ( !PyUnicode_Check(pyPath) )
        std::string path( PyUnicode_AsUTF8(pyPath) );
#if PY_MAJOR_VERSION < 3
    if ( !PyString_Check(pyPath)
        std::string path( PyString_AsString(pyPath) );
+#endif // #if/else PY_MAJOR_VERSION < 3